paper-badge.html

```diff
@@ -120,16 +120,15 @@
        * of the badge.
        */
       get target () {
-        var ownerRoot = Polymer.dom(this).getOwnerRoot();
         var parentNode = Polymer.dom(this).parentNode;
+        // If the parentNode is a document fragment, then we need to use the host.
+        var ownerRoot = Polymer.dom(this).getOwnerRoot();
         var target;
 
         if (this.for) {
-          target = parentNode.querySelector('#' + this.for);
-        } else if (parentNode.nodeType == 11) { // DOCUMENT_FRAGMENT_NODE
-          target = ownerRoot.host;
+          target = Polymer.dom(ownerRoot).querySelector('#' + this.for);
```

Comment:
Re: fallback. That's kiiinda maybe too magical and weird? Imagine debugging this: "it's getting attached to something, but what!". I don't love it :(